keydata.go

```diff
@@ -882,7 +676,7 @@ func (d *KeyData) WriteAtomic(w KeyDataWriter) error {
 
 // ReadKeyData reads the key data from the supplied KeyDataReader, returning a
 // new KeyData object.
-func ReadKeyData(r KeyDataReader) (*KeyData, error) {
+var ReadKeyData = func(r KeyDataReader) (*KeyData, error) {
```

Comment:
So this should no longer stand as these don't need to be mocked now that I uncoupled the legacy key tests from the new ones but still it is being used in several other places indeed.

For the first point, in most cases that's the issue, that these functions operate on non-exported symbols.
The second point is potentially a good idea, I will keep that in mind, thanks!
I had seen the third approach as well (and had forgotten about it) but I wasn't sure if that was the best way to do it anyway because it felt like a hack :)